bin/src/stats.rs

```diff
@@ -121,6 +121,11 @@ pub async fn get_qmdl_manifest(State(state): State<Arc<ServerState>>) -> Result<
     let qmdl_store = state.qmdl_store_lock.read().await;
     let mut entries = qmdl_store.manifest.entries.clone();
     let current_entry = qmdl_store.current_entry.map(|index| entries.remove(index));
+    if current_entry.clone().unwrap().has_warning {
+        info!("a heuristic triggered on this run!");
+        state.ui_update_sender.send(framebuffer::Color565::Red).await
+            .map_err(|e| (StatusCode::INTERNAL_SERVER_ERROR, format!("couldn't send ui update message: {}", e)))?;
+    }
```

Comment:
as per above, this should happen in the diag thread

Reply:
I moved this to analysis_writer.analyze() per your suggestion.